artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -1779,11 +1779,16 @@ public final SimpleString getLastValueProperty() {
 
    @Override
    public String toString() {
+      MessageDataScanningStatus scanningStatus = getDataScanningStatus();
+      Map<String, Object> applicationProperties = scanningStatus == MessageDataScanningStatus.SCANNED ?
+         getApplicationPropertiesMap(false) : Collections.EMPTY_MAP;
```

Comment:
This is probably the first one I could believe there might be a valid path to hitting, though I could also believe its again just a situation that we should enforce doesnt occur since it isnt expected to be possible in actual (vs broken test) usage.

Reply:
I see your point it should never happen but a toString should not change object state.